### PR TITLE
feat(shared): use onTextChange in the shared Rich text editor

### PR DESCRIPTION
**Pull request by celine-s**

### CODE REVIEW

File: packages/shared/src/components/RichTextEditor/types.ts [hunk 1]
```diff
@@ -10,12 +10,9 @@ export type RichTextEditorProps = {
...
     gap?: string;
     plugins?: PluginComposer;
-    updateValueOnChange?: boolean;
     showSerializedText?: boolean;
-    shouldPreventPageLeave?: boolean;
-    onBlur?: (value: string) => void;
-    onTextChange?: (value: string) => void;
-    onValueChanged?: (value: TreeOfNodes | null) => void;
+    updateValueOnChange?: boolean;
+    onTextChange: (value: string) => void;
```
Comment (Walkeezy) [inline packages/shared/src/components/RichTextEditor/types.ts:15]: Should this really be the only non-optional prop?

File: packages/text-block/src/TextBlock.tsx [hunk 2]
```diff
@@ -36,10 +25,7 @@ export const TextBlock = ({ appBridge }: BlockProps): ReactElement => {
             gap={gap}
             plugins={getPlugins(appBridge, columnNumber, gap)}
             placeholder={PLACEHOLDER}
-            onBlur={onTextChange}
-            onTextChange={onTextChange}
-            onValueChanged={() => setIsApiRequestPending(true)}
-            shouldPreventPageLeave={isApiRequestPending}
+            onTextChange={(value: string) => setBlockSettings({ content: value })}
```
Comment (ragi96) [inline packages/text-block/src/TextBlock.tsx:28]:
```suggestion
            onTextChange={(content: string) => setBlockSettings({ content })}
```

File: packages/shared/src/components/RichTextEditor/types.ts (hunk 1 above)
Comment (celine-s) [reply packages/shared/src/components/RichTextEditor/types.ts:15]: I see :D. I was just wondering if it ever can be undefined and I guessed "no". Didn't overthink the other ones, but know thinking about it maybe the id should be non-optional, but all others can be optional. Or what do you think?

File: packages/text-block/src/TextBlock.tsx (hunk 2 above)
Comment (celine-s) [reply packages/text-block/src/TextBlock.tsx]: good catch! Changed 👍

File: packages/asset-kit-block/src/AssetKitBlock.tsx [hunk 3]
```diff
@@ -131,7 +131,7 @@ export const AssetKitBlock = ({ appBridge }: BlockProps): ReactElement => {
                             }
                             isEditing={isEditing}
                             plugins={new PluginComposer({ noToolbar: true }).setPlugin()}
-                            onBlur={(value) => setBlockSettings({ buttonText: value })}
+                            onTextChange={(value) => setBlockSettings({ buttonText: value })}
```
[the remark this answers is not in the document]
Comment (celine-s) [reply packages/asset-kit-block/src/AssetKitBlock.tsx:134]: 👍

File: packages/shared/src/components/RichTextEditor/types.ts (hunk 1 above)
Comment (Walkeezy) [reply packages/shared/src/components/RichTextEditor/types.ts:15]: I think we could keep `onTextChange` optional too, then we would not need to adjust all the tests.

Comment (celine-s) [reply packages/shared/src/components/RichTextEditor/types.ts]: Changed 👍

**Review by ragi96 — COMMENTED** [review]
Let's wait for the final feedback of Schirmi

**Review by fulopdaniel — APPROVED** [review]

**Pull request merged**